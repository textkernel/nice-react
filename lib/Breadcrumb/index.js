export { default as BreadcrumbGroup } from './BreadcrumbGroup';
export { default as BreadcrumbItem } from './BreadcrumbItem';