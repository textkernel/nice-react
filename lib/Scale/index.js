export { default as ScaleGroup } from './ScaleGroup';
export { default as ScaleItem } from './ScaleItem';