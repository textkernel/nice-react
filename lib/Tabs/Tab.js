import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import niceProps from '../props';

const Tab = (props) => {
    const { headerText, isActive, onClick, tabIndex, icon, context } = props;
    const hasIcon = icon !== undefined;

    const onClickHandler = (event) => {
        event.preventDefault();
        onClick(tabIndex);
    };

    const tabIndexClass = `tab-link-class-${tabIndex}`;

    return (
        <li className={classNames({ current: isActive }, context)}>
            <a href="#" onClick={onClickHandler} className={classNames(tabIndexClass, { [`icon ${icon}`]: hasIcon })}>
                {headerText}
                {renderBadge(props)}
            </a>
        </li>
    );
};

function renderBadge(props) {
    if (!props.badge) {
        return;
    }

    return (
        <span className="nice-badge">
            {props.badge}
        </span>
    );
}

Tab.propTypes = {
    context: PropTypes.oneOf(niceProps.contexts),
    onClick: PropTypes.func,
    tabIndex: PropTypes.number,
    isActive: PropTypes.bool,
    icon: PropTypes.string,
    badge: PropTypes.string,
    headerText: PropTypes.string.isRequired
};

Tab.defaultProps = {
    context: 'default',
    onClick: null,
    tabIndex: 0,
    isActive: false,
    icon: null,
    badge: null,
    headerText: null,
};

export default Tab;