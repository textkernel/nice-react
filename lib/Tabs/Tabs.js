// https://medium.com/trisfera/a-simple-react-tabs-component-47cac2cfbb5
import React, { Component } from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export default class Tabs extends Component {

    static propTypes = {
        children: PropTypes.oneOfType([
            PropTypes.arrayOf(PropTypes.element),
            PropTypes.element,
        ]).isRequired,
        className: PropTypes.string,
        style: PropTypes.object,
        defaultActiveTabIndex: PropTypes.number,
        small: PropTypes.bool,
        wrapped: PropTypes.bool,
        onChangeTab: PropTypes.func
    };

    static defaultProps = {
        className: null,
        style: {},
        defaultActiveTabIndex: 0,
        onChangeTab: null,
        wrapped: false,
        small: false
    };

    constructor(props, context) {
        super(props, context);
        this.state = {
            activeTabIndex: this.props.defaultActiveTabIndex
        };
        this.handleTabClick = this.handleTabClick.bind(this);
    }

    handleTabClick(tabIndex) {
        this.setState({
            activeTabIndex: tabIndex === this.state.activeTabIndex 
                ? this.props.defaultActiveTabIndex 
                : tabIndex
        });
        if (this.props.onChangeTab !== undefined && typeof this.props.onChangeTab === 'function') {
            this.props.onChangeTab(tabIndex);
        }
    }

    // Encapsulate <Tabs/> component API as props for <Tab/> children
    renderChildrenWithTabsApiAsProps() {
        return React.Children.map(this.props.children, (child, index) =>
            React.cloneElement(child, {
                onClick: this.handleTabClick,
                tabIndex: index,
                isActive: index === this.state.activeTabIndex
            })
        );
    }

    // Render current active tab content
    renderActiveTabContent() {
        const { children } = this.props;
        const { activeTabIndex } = this.state;
        if (children[activeTabIndex]) {
            return children[activeTabIndex].props.children;
        }
        return null;
    }

    render() {
        const { wrapped, small, className, style } = this.props;

        return (
            <div className={classNames(className, { 'nice-tabs-wrapper': wrapped })} style={style}>
                <ul className={classNames('nice-tabs', { small }, 'bold')}>
                    {this.renderChildrenWithTabsApiAsProps()}
                </ul>
                <ul className="nice-tab-content">
                    {this.renderActiveTabContent()}
                </ul>
            </div>
        );
    }
}
