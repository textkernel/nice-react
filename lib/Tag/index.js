export { default as Tags } from './Tags';
export { default as Tag } from './Tag';