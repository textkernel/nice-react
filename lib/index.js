import '@textkernel/nice/dist/nice.min.css';

export { default as Panel } from './Panel';
export { Button, ButtonGroup } from './Buttons';
export { default as Progress } from './Progress';
export { default as Loading } from './Loading';
export { BreadcrumbGroup, BreadcrumbItem } from './Breadcrumb';