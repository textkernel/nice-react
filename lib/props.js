const contexts = ['brand', 'link', 'primary', 'accent', 'good', 'warning', 'bad'];
const sizes = ['small', 'medium', 'large'];
 
const props = {
  contexts,
  buttonContexts: contexts,
  buttonSizes: sizes 
};

module.exports = props;