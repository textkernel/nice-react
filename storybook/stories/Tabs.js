import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { withKnobsOptions, withKnobs, text, boolean, select, number } from '@storybook/addon-knobs';

import { Tabs, Tab } from 'nice-react';
import niceProps from '../../lib/props';

const stories = storiesOf('Tabs', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('Default', () => {
    const contextOptions = niceProps.contexts;

    const knobs = {
        small: boolean('Small', false),
        wrapped: boolean('Wrapped', false)
    }

    const tabItemKnobs = {
        context: select('Tab Context', contextOptions, niceProps.contexts[0]),
    }

    return (
        <Tabs {...knobs} onChangeTab={(tabIndex) => alert(`Tab ${tabIndex} selected!`)}>
            <Tab headerText="First" badge="56" icon="fa-home" {...tabItemKnobs}>
                <p>Tab number one</p>
            </Tab>
            <Tab headerText="Second Tab" icon="fa-angle-down" {...tabItemKnobs}>
                <p>Tab number two</p>
            </Tab>
            <Tab headerText="Third Tab" badge="4" {...tabItemKnobs}>
                <p>Tab number three</p>
            </Tab>
            <Tab headerText="Fourth Tab" {...tabItemKnobs}>
                <p>Tab number four</p>
            </Tab>
        </Tabs>
    );
});