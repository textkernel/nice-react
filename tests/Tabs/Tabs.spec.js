import React from 'react';
import { Tabs, Tab } from '../../lib/Tabs';

describe('<Tabs/>', () => {

    it('should render', () => {
        const Component = render(
            <Tabs>
                <Tab headerText="First" badge="56" icon="fa-home">
                    <p>Tab number zero</p>
                </Tab>
                <Tab headerText="Second" icon="fa-angle-down">
                    <p>Tab number one</p>
                </Tab>
            </Tabs>
        );
        expect(Component).toMatchSnapshot();
    });

    it('should render tab 0 content by default', () => {
        const Component = shallow(
            <Tabs>
                <Tab headerText="First" badge="56" icon="fa-home">
                    <p>Tab number zero</p>
                </Tab>
                <Tab headerText="Second" icon="fa-angle-down">
                    <p>Tab number one</p>
                </Tab>
            </Tabs>
        );
        expect(Component.find('.nice-tab-content').text()).toEqual('Tab number zero');
    });

    it("should allow to specify the 'defaultActiveTabIndex'", () => {
        const Component = shallow(
            <Tabs defaultActiveTabIndex={1}>
                <Tab headerText="First" badge="56" icon="fa-home">
                    <p>Tab number zero</p>
                </Tab>
                <Tab headerText="Second" icon="fa-angle-down">
                    <p>Tab number one</p>
                </Tab>
            </Tabs>
        );
        // Active tab content should be 1 instead of 0 now
        expect(Component.find('.nice-tab-content').text()).toEqual('Tab number one');
    });

    it('should change tab content to the selected tab', () => {
        const Component = mount(
            <Tabs defaultActiveTabIndex={0}>
                <Tab headerText="First" badge="56" icon="fa-home">
                    <p>Tab number zero</p>
                </Tab>
                <Tab headerText="Second" icon="fa-angle-down">
                    <p>Tab number one</p>
                </Tab>
            </Tabs>
        );
        expect(Component.find('.nice-tab-content').text()).toEqual('Tab number zero');
        // Select tab at index 1, and expect text to change
        Component.find('.tab-link-class-1').simulate('click', { preventDefault: () => { } });
        expect(Component.find('.nice-tab-content').text()).toEqual('Tab number one');
    });
});