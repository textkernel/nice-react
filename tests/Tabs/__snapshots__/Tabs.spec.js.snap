// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs/> should render 1`] = `
<div
  className="nice-tabs-wrapper"
>
  <ul
    className="nice-tabs bold"
  >
    <Tab
      badge="56"
      context="default"
      headerText="First"
      icon="fa-home"
      isActive={true}
      key=".0"
      onClick={[Function]}
      tabIndex={0}
    >
      <p>
        Tab number zero
      </p>
    </Tab>
    <Tab
      badge={null}
      context="default"
      headerText="Second"
      icon="fa-angle-down"
      isActive={false}
      key=".1"
      onClick={[Function]}
      tabIndex={1}
    >
      <p>
        Tab number one
      </p>
    </Tab>
  </ul>
  <ul
    className="nice-tab-content"
  >
    <p>
      Tab number zero
    </p>
  </ul>
</div>
`;
